src/vunnel/tool/fixdate/grype_db_first_observed.py

```diff
@@ -145,21 +146,63 @@ def _get_remote_digest(self, image_ref: str) -> str | None:
             self.logger.debug(f"failed to get remote digest: {e}")
             return None
 
+    def _resolve_image_ref(self, image_base: str) -> tuple[str, str | None]:
+        """resolve the image reference and digest, trying latest-zstd first then falling back to latest.
+
+        Returns:
+            Tuple of (image_ref, remote_digest) where remote_digest may be None if resolution failed.
+        """
+        # try latest-zstd first
+        image_ref = f"{image_base}:latest-zstd"
+        remote_digest = self._get_remote_digest(image_ref)
+        if remote_digest:
+            return image_ref, remote_digest
+
+        # fall back to latest
+        self.logger.debug("latest-zstd tag not resolvable, trying latest")
+        image_ref = f"{image_base}:latest"
+        remote_digest = self._get_remote_digest(image_ref)
+        return image_ref, remote_digest
+
+    def _pull(
+        self,
+        client: _ProgressLoggingOrasClient,
+        image_ref: str,
+        download_dir: Path,
+    ) -> None:
+        """pull the OCI artifact from the registry."""
+        self.logger.info(f"pulling fix date database from {image_ref}")
+        client.pull(target=image_ref, outdir=str(download_dir))
+        self.logger.info(f"successfully fetched fix date database for {self.provider}")
+
+    def _process_downloaded_file(self, download_zst_path: Path, download_db_path: Path) -> None:
+        """decompress zstd file if present, otherwise verify db file exists."""
+        if download_zst_path.exists():
+            self.logger.debug(f"decompressing {download_zst_path} to {download_db_path}")
+            dctx = zstandard.ZstdDecompressor()
+            with download_zst_path.open("rb") as ifh, download_db_path.open("wb") as ofh:
```

Comment:
Is there a path where the non-compressed `download_db_path` still exists from workspace cache of a previous run? If so, would this result in appending?